src/platform/plugins/shared/data_views/common/data_views/data_views.ts

```diff
@@ -1073,7 +1079,8 @@ export class DataViewsService {
   get = async (
     id: string,
     displayErrors: boolean = true,
-    refreshFields = false
+    refreshFields = false,
+    fieldSpecs?: FieldSpec[]
```

Comment:
This new parameter allows a client to supply the fields list for the data view, removing the need for a field fetch.